pandas/sparse/frame.py

```diff
@@ -302,7 +302,7 @@ def fillna(self, value=None, method=None, axis=0, inplace=False,
     # ----------------------------------------------------------------------
     # Support different internal representation of SparseDataFrame
 
-    def _sanitize_column(self, key, value):
+    def _sanitize_column(self, key, value, broadcast=False):
```

Comment:
So this was surprising to me: 
```
df1 = pd.DataFrame({'a': [1, 2, 3]})
df2 = pd.DataFrame({'b': [2,3,4]})
df = pd.concat([df1, df1, df2], axis=1).to_sparse()
df.index = [1, 2, 3]
df.loc[1, 'a'] = 3

## -- End pasted text --
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-e95308cf58e6> in <module>()
      3 df = pd.concat([df1, df1, df2], axis=1).to_sparse()
      4 df.index = [1, 2, 3]
----> 5 df.loc[1, 'a'] = 3

/Users/bkandel/code/forkz/pandas/pandas/core/indexing.pyc in __setitem__(self, key, value)
    138             key = com._apply_if_callable(key, self.obj)
    139         indexer = self._get_setitem_indexer(key)
--> 140         self._setitem_with_indexer(indexer, value)
    141 
    142     def _has_valid_type(self, k, axis):

/Users/bkandel/code/forkz/pandas/pandas/core/indexing.pyc in _setitem_with_indexer(self, indexer, value)
    545                 # scalar
    546                 for item in labels:
--> 547                     setter(item, value)
    548 
    549         else:

/Users/bkandel/code/forkz/pandas/pandas/core/indexing.pyc in setter(item, v)
    453 
    454             def setter(item, v):
--> 455                 s = self.obj[item]
    456                 pi = plane_indexer[0] if lplane_indexer == 1 else plane_indexer
    457 

/Users/bkandel/code/forkz/pandas/pandas/sparse/frame.pyc in __getitem__(self, key)
    345             return self._getitem_array(key)
    346         else:
--> 347             return self._get_item_cache(key)
    348 
    349     @Appender(DataFrame.get_value.__doc__, indents=0)

/Users/bkandel/code/forkz/pandas/pandas/core/generic.py in _get_item_cache(self, item)
   1384         if res is None:
   1385             values = self._data.get(item)
-> 1386             res = self._box_item_values(item, values)
   1387             cache[item] = res
   1388             res._set_as_cached(item, self)

/Users/bkandel/code/forkz/pandas/pandas/core/frame.py in _box_item_values(self, key, values)
   2391         items = self.columns[self.columns.get_loc(key)]
   2392         if values.ndim == 2:
-> 2393             return self._constructor(values.T, columns=items, index=self.index)
   2394         else:
   2395             return self._box_col_values(values, items)

AttributeError: 'BlockManager' object has no attribute 'T'
```
Looks like the sparse version doesn't deal with the broadcasting issue at all.